client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -844,34 +851,72 @@ export class PlanFeatures2023Grid extends Component<
 			} );
 	}
 
-	renderPlanStorageOptions( planPropertiesObj: PlanProperties[], options?: PlanRowOptions ) {
+	renderStorageAddOnDropdown( properties: PlanProperties ) {
+		const { planName, storageOptions, storageFeatures } = properties;
 		const { translate } = this.props;
+		const { selectedStorage } = this.state;
+
+		const selectControlOptions = storageOptions.map( ( storageOption ) => ( {
+			key: storageOption,
+			name: getStorageStringFromFeature( storageOption ) || '',
```

Comment:
Is `|| ''` just to match the type expected? Technically speaking, there can't be a case of "key" being defined and "name" being empty right? Maybe we can refactor with that in mind and only add an option when a value exists in full?

Reply:
[b0d1d2f](https://github.com/Automattic/wp-calypso/pull/79041/commits/b0d1d2fa3088689f84bcf0d307e627de34fd4791)